jdaviz/utils.py

```diff
@@ -436,41 +459,62 @@ def flux_conversion(values, original_units, target_units, spec=None, eqv=None, s
             values=values, orig_units=orig_units, targ_units=targ_units,
             eqv=eqv, image_data=image_data
             )
+    elif solid_angle_in_orig == solid_angle_in_targ == u.pix * u.pix:
+        # in the case where we have 2 SBs per solid pixel that need
+        # u.spectral_density equivalency, they can't be directly converted
+        # for whatever reason (i.e 'Jy / pix2' and 'erg / (Angstrom s cm2 pix2)'
+        # are not convertible). In this case, multiply out the factor of pix2 for
+        # conversion (same kind of thing _indirect_conversion is
+        # doing but we already know the exact angle units.
+        orig_units *= u.pix * u.pix
+        targ_units *= u.pix * u.pix
 
     return (values * orig_units).to_value(targ_units, equivalencies=eqv)
 
 
 def _indirect_conversion(values, orig_units, targ_units, eqv,
                          spec_unit=None, image_data=None):
+
+    # Note: is there a way we could write this to not require 'spec_unit'? It
+    # seems like it falls back on this to get a solid angle unit, but can we
+    # assume pix2 now if there are none? or use the display units?
```

Comment:
is this for the future or something we can sort out now?

Reply:
I think this should be a follow up, but I'll take a look and see if I can do it quickly in this ticket. This is related to passing all in-plugin unit conversions through flux_conversion rather than using astropy unit.to()